sources/api/early-boot-config/src/bin/early-boot-config.rs

```diff
@@ -104,39 +187,34 @@ async fn run() -> Result<()> {
 
     info!("early-boot-config started");
 
-    info!("Retrieving platform-specific data");
-    let uri = &format!(
-        "{}?tx={}",
-        constants::API_SETTINGS_URI,
-        constants::LAUNCH_TRANSACTION
-    );
-    let method = "PATCH";
-    for settings_json in Platform
-        .platform_data()
-        .await
-        .context(error::ProviderSnafu)?
-    {
-        // Don't send an empty request to the API
-        if settings_json.json.is_empty() {
-            warn!("{} was empty", settings_json.desc);
+    info!("Gathering user data providers");
+    let providers = gather_providers()?;
+    for provider in providers {
+        debug!("Found '{}', running it...", provider.display());
+        let result = run_provider(&args.log_level, &provider).await?;
```

Comment:
I think we should probably run the providers in parallel, but then apply the results in-order.